divide_and_conquer/mergesort.py

```diff
@@ -1,48 +1,58 @@
-def merge(arr, left, mid, right):
-    # overall array will divided into 2 array
-    # left_arr contains the left portion of array from left to mid
-    # right_arr contains the right portion of array from mid + 1 to right
-    left_arr = arr[left : mid + 1]
-    right_arr = arr[mid + 1 : right + 1]
-    k = left
-    i = 0
-    j = 0
-    while i < len(left_arr) and j < len(right_arr):
-        # change sign for Descending order
-        if left_arr[i] < right_arr[j]:
-            arr[k] = left_arr[i]
-            i += 1
+def merge(left_half, right_half):
```

Comment:
Please use type hints for all functions and parameters.

Refer: https://www.python.org/dev/peps/pep-0484/
Or: https://docs.python.org/3/library/typing.html

Reply:
This was new to me. I did not know about type hints... Thanks for the suggestion